pkg/repository/product/productRepositoryImpl.go

```diff
@@ -83,12 +95,42 @@ func (r *productRepositoryImpl) Delete(ctx context.Context, id string) (*entitie
 		return nil, errors.New(fmt.Sprintf("failed to delete product"))
 	}
 
-	return product.ToProductEntity(), nil
+	productEntity := product.ToProductEntity()
+	r.SetProductSubAttributes(product, sp)
+	return productEntity, nil
 }
 
-func ToProductModel(e *entities.Product) *model.Product {
+func (r *productRepositoryImpl) ToProductModel(e *entities.Product) *model.Product {
 	return &model.Product{
 		Name:  e.ProductName,
 		Price: e.ProductPrice,
 	}
 }
+
+func (r *productRepositoryImpl) SetProductSubAttributes(productData any, sp trace.Span) {
```

Comment:
I think it can use switch case for readability and easy to maintain

Reply:
OK, i rarely use Switch case Thx